docs/deprecated.md

```diff
@@ -281,19 +281,39 @@ When using cgroups v2, the `--blkio-weight` options are implemented using
 ### Kernel memory limit
 
 **Deprecated in Release: v20.10**
+**Removed in Release: v22.04**
 
-Specifying kernel memory limit (`docker run --kernel-memory`) is now marked as deprecated,
-as [Linux kernel deprecated `kmem.limit_in_bytes` in v5.4](https://github.com/torvalds/linux/commit/0158115f702b0ba208ab0b5adf44cae99b3ebcc7).
+Specifying kernel memory limit (`docker run --kernel-memory`) is no longer supported
+because the [Linux kernel deprecated `kmem.limit_in_bytes` in v5.4](https://github.com/torvalds/linux/commit/0158115f702b0ba208ab0b5adf44cae99b3ebcc7).
+The OCI runtime specification now marks this option (as well as `--kernel-memory-tcp`)
+as ["NOT RECOMMENDED"](https://github.com/opencontainers/runtime-spec/pull/1093),
+and OCI runtimes such as `runc` no longer support this option.
+
+Docker API v1.42 and up now ignores this option when set. Older versions of the
+API continue to accept the option, but depending on the OCI runtime used, may
+take no effect.
+
+> **Note**
+> 
+> While not deprecated (yet) in Docker, the OCI runtime specification also
+> deprecated the `memory.kmem.tcp.limit_in_bytes` option. When using `runc` as
+> runtime, this option takes no effect. The linux kernel did not explicitly
+> deprecate this feature, and there is a tracking ticket in the `runc` issue
+> tracker to determine if this option should be reinstated or if this was an
+> oversight of the Linux kernel maintainers (see [opencontainers/runc#3174](https://github.com/opencontainers/runc/issues/3174)).
+> 
+> The `memory.kmem.tcp.limit_in_bytes` option is only supported with cgroups v1,
+> and not available on installations running with cgroups v2. This option is
+> only supported by the API, and not exposed on the `docker` command-line.
```

Comment:
LOL, so I went looking for the `--kernel-memory-tcp` flag, but it looks like we never exposed this option on the CLI (perhaps through compose or swarm though?)